### PR TITLE
[CCI] Add bluebird replaces for src/plugins/saved_objects

### LINKED ISSUE #3662: Eliminate Bluebird in favor of native async functionality
Bluebird is a open-source JavaScript library for working with asynchronous code. It is a Promise library that provides a number of useful features and enhancements to the built-in JavaScript Promise object, including performance optimizations and additional methods.

The task is to replace all instances of Bluebird with native async functionality in the codebase. This should be done using the async and await keywords, which have been natively available for over 5 years in all relevant environments. Promises have also been available for longer.

- [ ] Identify all instances where Bluebird is used in the codebase
- [ ] Replace the Bluebird calls with the equivalent native async functionality using async and await
   - [ ] Below are the list of bluebird functions that are currently used and their equivalent native implementation
```
         all -> Promise.all
         settle -> Promise.allSettled
         resolve -> Promise.resolve
         reject-> Promise.reject
         setSchedular -> queueMicrotask
         fromNode ->  const util = require('util');
                                await util.promisify(readFileCallback);
         delay ->  function delay(ms) {
                        return new Promise(resolve => setTimeout(resolve, ms));}
         mapAsync -> function mapAsync(array, mapper) {
                               return Promise.all(array.map(mapper));}
         promisify -> await util.promisify(fs.readFile);
```
- [ ] Test the updated code to ensure that it is functioning as expected

### PR DESCRIPTION
**Pull request by Nicksqain**
### Issues Resolved

- closes #3662 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/saved_objects/public/saved_object/saved_object.test.ts [hunk 1]
```diff
@@ -246,7 +245,7 @@ describe('Saved Object', () => {
         return createInitializedSavedObject({ type: 'dashboard' }).then((savedObject) => {
           savedObjectsClientStub.create = jest.fn().mockImplementation(() => {
             expect(savedObject.isSaving).toBe(true);
-            return Bluebird.reject('');
+            return Promise.reject();
```
Comment (ananzh) [inline src/plugins/saved_objects/public/saved_object/saved_object.test.ts:248]: What is the difference between reject a promise `''` in Bluebird and here `reject()`?

Comment (Nicksqain) [reply src/plugins/saved_objects/public/saved_object/saved_object.test.ts:248]: There is no difference, it's just that bluebird's first parameter is the reject reason. And it is mandatory. So I just removed the message, as it simply does not exist

**Review by ananzh — APPROVED** [review]

**Review by ananzh — DISMISSED** [review]

**Review by AMoo-Miki — APPROVED** [review]
Thanks.

**Comment by ananzh** [comment]
@Nicksqain This PR is ready. Could you add it to our CHANGELOG? After that, we could approve again and merge in.

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**